besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1719,6 +1718,8 @@ private JsonRpcConfiguration jsonRpcConfiguration() {
           "Unable to authenticate JSON-RPC HTTP endpoint without a supplied credentials file or authentication public key file");
     }
 
+    removeRpcApisForDisabledFeatures();
```

Comment:
Rather than silently removing I think this should be a CLI failure.  Much like specifying apis without enabling http-rpc is an error adding a API for a feature that is not enabled is an error.

Reply:
that would be a breaking change though for docker setups?

Reply:
It would be, but so is silently disabling the APIs because of flag changes.